README.rst

```diff
@@ -161,20 +160,29 @@ The tests are also run automatically with `Travis CI <https://travis-ci.org/RaRe
 Supported Compression Formats
 -----------------------------
 
-``smart_open`` allows reading and writing gzip, bzip2 and xz files.
+``smart_open`` allows reading and writing gzip and bzip2 files.
 They are transparently handled over HTTP, S3, and other protocols, too, based on the extension of the file being opened.
-You can easily add support for other file extensions and compression formats:
+You can easily add support for other file extensions and compression formats.
+For example, to open xz-compressed files:
 
 .. code-block:: python
 
-    def _handle_lzma(file_obj, mode):
-        import lzma
-        return lzma.LZMAFile(filename=file_obj, mode=mode, format=lzma.FORMAT_ALONE)
-
+    import lzma, os
     from smart_open import open, register_compressor
-    register_compressor('.lzma', _handle_lzma)
-    with open('file.lzma', ...) as fin:
-        pass
+
+    def _handle_xz(file_obj, mode):
+        return lzma.LZMAFile(filename=file_obj, mode=mode, format=lzma.FORMAT_XZ)
+
+    register_compressor('.xz', _handle_xz)
+
+    data_path = './smart_open/tests/test_data/crime-and-punishment.txt.xz'
+    with open(data_path) as f:
+        crime_and_punishment = f.read()
```

Comment:
```suggestion
        ... text = fin.read()
        ... print(len(text))
        1234 
```

Replace 1234 with the actual number of read characters.

You can then run "python -m doctest README.rst". It will run the example above, and compare the read number of bytes against the expected value (1234 in this case).